src/com/etendoerp/copilot/process/CheckHostsButton.java

```diff
@@ -0,0 +1,215 @@
+        String copilotHost = CopilotUtils.getCopilotHost();
+        String copilotPort = CopilotUtils.getCopilotPort();
+
+        HttpURLConnection pythonConnection = null;
+        try {
+            URL pythonUrl = new URL("http://" + copilotHost + ":" + copilotPort + "/checkCopilotHost");
+            pythonConnection = (HttpURLConnection) pythonUrl.openConnection();
+            pythonConnection.setRequestMethod("POST");
+            pythonConnection.setDoOutput(true);
+            pythonConnection.setRequestProperty("Authorization", "Bearer " + token);
+            pythonConnection.setRequestProperty("Content-Type", CONTENT_TYPE);
+            pythonConnection.setRequestProperty("Accept", CONTENT_TYPE);
+
+            try (BufferedReader reader = new BufferedReader(
+                new InputStreamReader(pythonConnection.getInputStream()))) {
+                int responsePythonCode = pythonConnection.getResponseCode();
+                if (responsePythonCode == HttpServletResponse.SC_OK) {
+                    log4j.info(COPILOT_HOST_SUCCESS);
+                    result.put(COPILOT_HOST, COPILOT_HOST_SUCCESS);
+                    StringBuilder responseBuilder = new StringBuilder();
+                    String line;
+                    while ((line = reader.readLine()) != null) {
+                        responseBuilder.append(line);
+                    }
+                    String responseBody = responseBuilder.toString();
+
+                    if (StringUtils.contains(responseBody, ETENDO_HOST_DOCKER_SUCCESS)) {
+                        log4j.info(ETENDO_HOST_DOCKER_SUCCESS);
+                        result.put(ETENDO_HOST_DOCKER, ETENDO_HOST_DOCKER_SUCCESS);
+                    } else {
+                        log4j.error(CheckHostsButton.ERROR_ETENDO_HOST_DOCKER);
+                        result.put(ETENDO_HOST_DOCKER, CheckHostsButton.ERROR_ETENDO_HOST_DOCKER);
+                    }
+                } else {
+                    result.put(COPILOT_HOST, ERROR_COPILOT_HOST + ERROR + responsePythonCode);
+                    result.put(ETENDO_HOST_DOCKER, ERROR_ETENDO_HOST_DOCKER);
+                }
+            }
+
+        } catch (Exception e) {
+            if (StringUtils.contains(e.getMessage(), "Connection refused")) {
+                String message = "Connection refused. Is Copilot up?";
+                log4j.error(ERROR_COPILOT_HOST + " {}", message);
+                result.put(COPILOT_HOST, ERROR_COPILOT_HOST + ERROR + message);
+            } else {
+                result.put(COPILOT_HOST, ERROR_COPILOT_HOST + ERROR + e);
+                log4j.error(ERROR_COPILOT_HOST + " ", e);
+            }
+            result.put(ETENDO_HOST_DOCKER, ERROR_ETENDO_HOST_DOCKER);
+            log4j.error(ERROR_ETENDO_HOST_DOCKER);
+        } finally {
+            if (pythonConnection != null) {
+                pythonConnection.disconnect();
+            }
+        }
+    }
+
```

Comment:
This 2 functions do a post requests, we can unify the logics in a function. I dont remember if exists a util for this already(?


```suggestion
/**
 * Creates and configures an HttpURLConnection for a given URL and token.
 *
 * @param urlString The URL to connect to.
 * @param token     The security token for authentication.
 * @return Configured HttpURLConnection.
 * @throws IOException If an error occurs while creating the connection.
 */
private HttpURLConnection createConnection(String urlString, String token) throws IOException {
    URL url = new URL(urlString);
    HttpURLConnection connection = (HttpURLConnection) url.openConnection();
    connection.setRequestMethod("POST");
    connection.setDoOutput(true);
    connection.setRequestProperty("Authorization", "Bearer " + token);
    connection.setRequestProperty("Content-Type", CONTENT_TYPE);
    connection.setRequestProperty("Accept", CONTENT_TYPE);
    return connection;
}

/**
 * Checks the Etendo host connectivity and configuration.
 *
 * @param token  The security token used for authentication.
 * @param result The JSON object to store the results of the check.
 */
private void checkEtendoHost(String token, JSONObject result) {
    String etendoHost = CopilotUtils.getEtendoHost();
    HttpURLConnection connection = null;
    try {
        connection = createConnection(etendoHost + "/sws/copilot/configcheck", token);

        int responseCode = connection.getResponseCode();
        String etendoHostMessage;
        if (responseCode == 200) {
            log4j.info(ETENDO_HOST_SUCCESS);
            etendoHostMessage = ETENDO_HOST_SUCCESS;
        } else {
            etendoHostMessage = ERROR_ETENDO_HOST + ": Error " + responseCode;
            log4j.error(ERROR_ETENDO_HOST);
        }
        result.put(ETENDO_HOST, etendoHostMessage);
    } catch (Exception e) {
        log4j.error(ERROR_ETENDO_HOST + ": ", e);
    } finally {
        if (connection != null) {
            connection.disconnect();
        }
    }
}

/**
 * Checks the Copilot host connectivity and configuration.
 *
 * @param token  The security token used for authentication.
 * @param result The JSON object to store the results of the check.
 */
private void checkCopilotHost(String token, JSONObject result) {
    String copilotHost = CopilotUtils.getCopilotHost();
    String copilotPort = CopilotUtils.getCopilotPort();
    HttpURLConnection pythonConnection = null;
    try {
        pythonConnection = createConnection("http://" + copilotHost + ":" + copilotPort + "/checkCopilotHost", token);

        try (BufferedReader reader = new BufferedReader(
                new InputStreamReader(pythonConnection.getInputStream()))) {
            int responsePythonCode = pythonConnection.getResponseCode();
            if (responsePythonCode == HttpServletResponse.SC_OK) {
                log4j.info(COPILOT_HOST_SUCCESS);
                result.put(COPILOT_HOST, COPILOT_HOST_SUCCESS);

                StringBuilder responseBuilder = new StringBuilder();
                String line;
                while ((line = reader.readLine()) != null) {
                    responseBuilder.append(line);
                }
                String responseBody = responseBuilder.toString();

                if (StringUtils.contains(responseBody, ETENDO_HOST_DOCKER_SUCCESS)) {
                    log4j.info(ETENDO_HOST_DOCKER_SUCCESS);
                    result.put(ETENDO_HOST_DOCKER, ETENDO_HOST_DOCKER_SUCCESS);
                } else {
                    log4j.error(CheckHostsButton.ERROR_ETENDO_HOST_DOCKER);
                    result.put(ETENDO_HOST_DOCKER, CheckHostsButton.ERROR_ETENDO_HOST_DOCKER);
                }
            } else {
                result.put(COPILOT_HOST, ERROR_COPILOT_HOST + ERROR + responsePythonCode);
                result.put(ETENDO_HOST_DOCKER, ERROR_ETENDO_HOST_DOCKER);
            }
        }

    } catch (Exception e) {
        if (StringUtils.contains(e.getMessage(), "Connection refused")) {
            String message = "Connection refused. Is Copilot up?";
            log4j.error(ERROR_COPILOT_HOST + " {}", message);
            result.put(COPILOT_HOST, ERROR_COPILOT_HOST + ERROR + message);
        } else {
            result.put(COPILOT_HOST, ERROR_COPILOT_HOST + ERROR + e);
            log4j.error(ERROR_COPILOT_HOST + " ", e);
        }
        result.put(ETENDO_HOST_DOCKER, ERROR_ETENDO_HOST_DOCKER);
        log4j.error(ERROR_ETENDO_HOST_DOCKER);
    } finally {
        if (pythonConnection != null) {
            pythonConnection.disconnect();
        }
    }
}
```